### ISSUE TITLE
`select ![]` after `from_text` sometimes causes panic

### ISSUE THREAD

**Issue by eitsupi** [PRQL/prql#2217]
### What happened?

The following query seems to cause panic.

### PRQL input

```elm
from_text format:json '[{"a":1, "b":1}]'
loop (
    derive b_new = a + b
    select [a=b, b=b_new]
)
select ![b]
take 2
```

### SQL output

```SQL
NA
```

### MVCE confirmation

- [X] Minimal example
- [X] New issue

**Comment by Jelenkee** [comment]
It's not a problem with loop. Even that snippet causes an error.
```
from_text format:json '[{"a":1, "b":1}]'
select ![b]
```

**Comment by eitsupi** [comment]
> It's not a problem with loop. Even that snippet causes an error.
> 
> ```
> from_text format:json '[{"a":1, "b":1}]'
> select ![b]
> ```

Thanks for pointing that.
This also panic at the same point.

```sh
$ echo 'from_text format:json \'[{"a":1, "b":1}]\'
  select ![b]' | prqlc compile
The application panicked (crashed).
Message:  called `Option::unwrap()` on a `None` value
Location: /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/prql-compiler-0.6.1/src/semantic/transforms.rs:705
```

**Comment by aljazerzen** [comment]
I believe this was fixed with #2399
